Task/Task_Management_System/views.py

```diff
@@ -0,0 +1,236 @@
+
+            if User.objects.filter(email=email).exists():
+                errors["email"] = "Email is already registered."
+
+            if "@" not in email or "." not in email:
+                errors["email"] = "Enter a valid email address."
+
+            if len(password) < 5:
+                errors["password"] = "Password must be at least 5 characters."
+
+            if not re.search(r"[!@#$%^&*|<>,(){}]", password):
+                errors["password"] = "Password must contain at least one special character."
+
+            if errors:
+                return render(request, "signup.html", {"errors": errors})
+
+            user = User(first_name=username, email=email)
+            user.set_password(password)
+            user.save()
+
+            return redirect("login")
+        except IntegrityError as e:
+            return render(request, 'signup.html', {'error': 'Email or Username already exists.', 'details': str(e)})
+        except Exception as e:
+            return render(request, 'signup.html', {'error': 'An error occurred', 'details': str(e)})
+
+
+class Login(View):
+    def get(self, request):
+        return render(request, 'login.html')
+
+    def post(self, request):
+        email = request.POST.get('email')
+        password = request.POST.get('password')
+        user = authenticate(request, username=email, password=password)
+
+        if user:
+            login(request, user)
+            return redirect('home_page')
+        return render(request, 'login.html', {'error': 'Invalid email or password'})
+
+
+class My_Dashboard(LoginRequiredMixin, View):
+    def get(self, request, *args, **kwargs):
+        data = get_object_or_404(User, first_name=request.user)
+        tasks = Task.objects.filter(assigned_to=request.user)
+        print('Username:', data)
+        print("TASKS:", tasks)
+        task_status_count = {
+            'Pending': 0,
+            'Completed': 0,
+        }
+        for task in tasks:
+            if task.status == 'Pending':
+                task_status_count['Pending'] += 1
+            elif task.status == 'Completed':
+                task_status_count['Completed'] += 1
+        return render(request, 'my_dashboard.html', {'tasks': tasks, 'task_status_count': task_status_count, 'data': data})
+
+
+class Show_Profile(LoginRequiredMixin, View):
+    def get(self, request):
+        data = get_object_or_404(User, email=request.user.email)
+        return render(request, 'show_profile.html', {'data': data})
+
+
+class Create_Task(LoginRequiredMixin, View):
+    login_url = 'login'
+
+    def get(self, request):
+        form = TaskForm()
+        users = User.objects.all()
+        return render(request, 'create_task.html', {'form': form, 'users': users})
+
+    def post(self, request):
+        form = TaskForm(request.POST)
+        if form.is_valid():
+            title = form.cleaned_data['title']
+            description = form.cleaned_data['description']
```

Comment:
Please improve code, try to use form.save().
And for email create a function in utils.py then use here